src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/Helper.Win32Unix.cs

```diff
@@ -51,11 +52,33 @@ internal static void GetDefaultIdentityAndHash(out object identity, out string h
             Assembly? assembly = Assembly.GetEntryAssembly();
 
             if (assembly == null)
+            {
                 throw new IsolatedStorageException(SR.IsolatedStorage_Init);
+            }
+
+            // in case of SingleFile, Location.Length is 0, we will use native entrypoint's filename.
+            if (assembly.Location.Length == 0)
```

Comment:
Are you sure the Lenght is 0 only for coreclr single-file?

Reply:
I was checking the (magic) string `Assembly.GetEntryAssembly().ManifestModule.Name == "<Unknown>"` before (new in 5.0), which is currently only returned for single file. I was suggested to change to this approach instead in: https://github.com/dotnet/runtime/pull/42176#discussion_r487533410. If there is a better way to detect a single file, can use that instead.

Reply:
The length is 0 for assembly loaded from memory that is not physically present on disk. It is the case of coreclr single file, but it can happen in other scenarios as well. For example, I believe Blazor apps are also loading the assemblies from memory and will take this path.

Reply:
This assembly is unsupported on blazorwasm, in latest (as of this moment's) RC2 build, we get:

> blazor.webassembly.js:1 System.PlatformNotSupportedException: System.IO.IsolatedStorage is not supported on this platform.

Reply:
I was questioning the general approach. The single-exe is only one of many bundling techniques which publishing will ship in .net5 or later. Maybe we should think about adding specific APIs to detect that instead of using `assembly.Location.Length == 0` and hope it won't be hit for example on Android.

Could we add a test for this change to see it works and does not break existing configurations?

Reply:
AFAIK, currently there is no proper API to detect SingleFile. I agree that it should have been part of the SingleFile story for 5.0. The workaround is to use extract-to-disk and this "fix" is to prevent the user for opting into the workaround specifically to avoid the potential regression.

There was some discussion about the test infra here: https://github.com/dotnet/runtime/issues/42161, currently libraries infra does not have facility to test SingleFile either. The workaround for that is to write unit tests for these in the existing installer test project, dedicated for SingleFile. Is that ok to add a test for it in, for example, https://github.com/dotnet/runtime/blob/master/src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/SingleFileApiTests.cs (for now; until libraries get singlefile test ability)?

Generally, there is test coverage for IsolatedStorageFile non-singlefile in this repository, which covers the currently supported platforms and the CI is passing.

Reply:
> no API to detect SingleFile

This was intentional design decision. SingleFile can come in many variants. We have shipped one variant .NET 3, shipping a different variant in .NET 5 and we will likely see more variants in future.

The APIs for this scenario should return concrete properties, like the proposed API for process executable path.